archethic/contracts/token_pool.exs

```diff
@@ -128,15 +128,18 @@ condition triggered_by: transaction, on: request_secret_hash(htlc_genesis_addres
     )
 ]
 
-actions triggered_by: transaction, on: request_secret_hash(htlc_genesis_address, amount, _user_address, chain_id) do
+actions triggered_by: transaction, on: request_secret_hash(htlc_genesis_address, amount, _user_address, chain_id, evm_user_address) do
   # Here delete old secret that hasn't been used before endTime
   requested_secrets = State.get("requested_secrets", Map.new())
 
   secret = Crypto.hmac(transaction.address)
   secret_hash = Crypto.hash(secret, "sha256")
 
-  # Perform a first hash to combine data and chain_id
-  abi_data = Evm.abi_encode("(bytes32, bytes32,uint)", [Crypto.hash(htlc_genesis_address), secret_hash, chain_id])
+  protocol_fee = Contract.call_function(@FACTORY_ADDRESS, "get_protocol_fee", [])
+  fee_amount = amount * (protocol_fee / 100)
+  evm_amount = Math.trunc((amount - fee_amount) * 1.0e18)
```

Comment:
Here you should not use 1.0e18 directly but use the "decimals" key from the map returned by `get_chain_data`